Add explicit re-export for werkzeug.routing

alisaifee opened the pull request:
`werkzeug.routing` in werkzeug < 2.2 was a single file submodule and thus all the exports were explicit. Since 2.2.x all the exported attributes are implicit re-exports
and can get flagged as typing errors (for example with mypy). This adds the exported classes and methods to the `__all__` variable of `werkzeug.routing`

- fixes #2492

🚧 I've left some of the checklist items until a preliminary review since I'm not sure if this is even important enough for the project and whether it needs to be in the changelog etc..

Checklist:

- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.

pgjones commented:
Pallets projects prefer the `import x as x` style, see the [base init](https://github.com/pallets/werkzeug/blob/main/src/werkzeug/__init__.py)

alisaifee commented:
> Pallets projects prefer the `import x as x` style, see the [base init](https://github.com/pallets/werkzeug/blob/main/src/werkzeug/__init__.py)

Thanks! updated in 1b1ddacd29a9dc7c6ba6ef9e8e5fca8a23374d51

pgjones commented:
I've changed the base to 2.2.x (as a bug fix for typing), could you update the branch?

alisaifee commented:
> I've changed the base to 2.2.x (as a bug fix for typing), could you update the branch?

👍 I've squashed and updated the commit description as well.

pgjones commented:
Could you also add a short changelog linking to the issue? (I was in two minds about this, but I think it is worthwhile).

pgjones commented:
Thanks
